### PR TITLE
simple changelog function

### PR DESCRIPTION
**Pull request by jbpratt**
With the auto-deploy of 'chat-gui', which is all this will run for (for
now) will always be up to date locally. Open the git repo locally, get
the last 3 commits, mutilate them, then write to chat

The whitespace/newline stripping can absolutely be done better, but I was
lazy. And it might be best to cache recent results but this seems like a
stretch goal. I'm not picking or looking for any commit name in
particular so we will get all.

My idea is on merge of commit (squash on the PR), set a
descriptive but simple message and this will be used and not end up
obnoxious. These should be somewhat comprehensible to the average
chatter

The test repo is archived so I assume it is fine for testing, it would
be difficult for anything there to change. Welcome to a better idea
though. It is also whitespace sensitive, which I really dislike.

### CODE REVIEW

**Review by tensei — APPROVED** [review]

**Comment by slugalisk** [comment]
It's not realistic to expect that commit messages will be readily accessible to people not familiar with the projects or that every commit is relevant to chatters. For this to be useful there needs to be a way to explicitly set the bot text when it's appropriate. There is an existing git convention for extended commit summaries so there is no reason to munge together some frankenmessage from multiple commits...

The first line of the commit message should be ignored completely. The bot should look for `CHANGELOG:` in the full commit message then use whatever follows. This is similar to the way github allows closing issues automatically when diffs land with `fixes:#id`.

**Comment by jbpratt** [comment]
> It's not realistic to expect that commit messages will be readily accessible to people not familiar with the projects or that every commit is relevant to chatters. For this to be useful there needs to be a way to explicitly set the bot text when it's appropriate. There is an existing git convention for extended commit summaries so there is no reason to munge together some frankenmessage from multiple commits...
> 
> The first line of the commit message should be ignored completely. The bot should look for `CHANGELOG:` in the full commit message then use whatever follows. This is similar to the way github allows closing issues automatically when diffs land with `fixes:#id`.

More than happy to implement this if we will actually follow it. My initial request was to simply do what you are asking in the first line of the commit message it self, but we can go with the extended summary. Like `bug:` or `fix:`, `feature:` then allowing me to simply grab the first line, which on merge, should be explanatory. 

But I am fine to do the extended message, just going forward, don't merge PRs without it.

Edit:
Something like this was my initial thought https://github.com/conventional-changelog/commitlint

File: changelog.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		return nil, fmt.Errorf("failed 'git log': %v", err)
+	}
+
+	for i := 0; i < n; i++ {
+		com, err := cIter.Next()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get next commit log: %v", err)
+		}
+
+		if strings.Contains(com.Message, "CHANGELOG") {
```
Comment (slugalisk) [inline changelog.go]: missing colon? it looks like omitting the : on accident would cause panic?

Comment (jbpratt) [reply changelog.go]: I was missing the colon. Added that, but how would it have caused a panic? Contains matches substrings, so I think it still would have worked fine, at least I had no issues manually running it. I will get the test in soon.

**Pull request merged**